examples/static_meshlets/source/static_meshlets.cpp

```diff
@@ -0,0 +1,497 @@
+				allMatConfigs.push_back(pair.first);
+			}
+
+			for (size_t mpos = 0; mpos < meshIndicesInOrder.size(); mpos++) {
+				auto meshIndex = meshIndicesInOrder[mpos];
+				std::string meshname = curModel->name_of_mesh(mpos);
+
+				auto texelBufferIndex = dataForDrawCall.size();
+				auto& drawCallData = dataForDrawCall.emplace_back();
+
+				drawCallData.mMaterialIndex = static_cast<int32_t>(matOffset);
+				drawCallData.mModelMatrix = globalTransform * curModel->transformation_matrix_for_mesh(meshIndex);
+				// Find and assign the correct material (in the ~"global" allMatConfigs vector!)
+				for (auto pair : distinctMaterials) {
+					if (std::end(pair.second) != std::find(std::begin(pair.second), std::end(pair.second), meshIndex)) {
+						break;
+					}
+
+					drawCallData.mMaterialIndex++;
+				}
+
+				auto selection = gvk::make_models_and_meshes_selection(curModel, meshIndex);
+				std::vector<gvk::mesh_index_t> meshIndices;
+				meshIndices.push_back(meshIndex);
+				// Build meshlets:
+				std::tie(drawCallData.mPositions, drawCallData.mIndices) = gvk::get_vertices_and_indices(selection);
+				drawCallData.mNormals = gvk::get_normals(selection);
+				drawCallData.mTexCoords = gvk::get_2d_texture_coordinates(selection, 0);
+
+				// create selection for the meshlets
+				std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<gvk::mesh_index_t>>> meshletSelection;
+				meshletSelection.emplace_back(avk::shared(curModel), meshIndices);
+
+				auto cpuMeshlets = gvk::divide_into_meshlets(meshletSelection);
+#if USE_DIRECT_MESHLET
+				gvk::serializer serializer("direct_meshlets-" + meshname + "-" + std::to_string(mpos) + ".cache");
+				auto [gpuMeshlets, _] = gvk::convert_for_gpu_usage_cached<gvk::meshlet_gpu_data<sNumVertices, sNumIndices>, sNumVertices, sNumIndices>(serializer, cpuMeshlets);
+#else
+				gvk::serializer serializer("indirect_meshlets-" + meshname + "-" + std::to_string(mpos) + ".cache");
+				auto [gpuMeshlets, generatedMeshletData] = gvk::convert_for_gpu_usage_cached<gvk::meshlet_redirected_gpu_data, sNumVertices, sNumIndices>(serializer, cpuMeshlets);
+				drawCallData.mMeshletData = std::move(generatedMeshletData.value());
+#endif
+
+				serializer.flush();
+
+				// fill our own meshlets with the loaded/generated data
+				for (size_t mshltidx = 0; mshltidx < gpuMeshlets.size(); ++mshltidx) {
+					auto& genMeshlet = gpuMeshlets[mshltidx];
+
+					auto& ml = meshletsGeometry.emplace_back(meshlet{});
+
+#pragma region start to assemble meshlet struct
+					ml.mTransformationMatrix = drawCallData.mModelMatrix;
+					ml.mMaterialIndex = drawCallData.mMaterialIndex;
+					ml.mTexelBufferIndex = static_cast<uint32_t>(texelBufferIndex);
+
+					ml.mGeometry = genMeshlet;
+#pragma endregion 
+				}
+			}
+		} // end for (auto& loadInfo : toLoad)
+
+		// lambda to create all the buffers for us
+		auto addDrawCalls = [this](auto& dataForDrawCall, auto& drawCallsTarget) {
```

Comment:
I think, it would be less confusing, or easier to read/comprehend for new framework users if this was a method `static_meshlets_app::add_draw_calls` instead of a local lambda within `static_meshlets_app::initialize`. Please refactor!